### PR TITLE
Electron and react-i18next dependencies update

### PR DESCRIPTION
**Pull request by rihardsgravis**
# Description

- Update Trinity desktop Electron version to 3.0.3 and Node 10 build support.
- Update react-i18next to 8.0.7 and move i18n initialisation to shared modules

## Type of change

- Enhancement (a non-breaking change which adds functionality)

# How Has This Been Tested?

Tested on macOS; Ubuntu and iOS

# Checklist:

- [x] My code follows the style guidelines for this project
- [x] I have performed a self-review of my own code
- [x] New and existing unit tests pass locally with my changes
- [x] For changes to `mobile` that include native code (including React Native modules): I have verified that both iOS and Android successfully build in both `Debug` and `Release` modes
- [x] For changes to `shared`: If applicable, I have verified that my changes are implemented correctly in `desktop` and `mobile`

### CODE REVIEW

File: src/mobile/src/ui/components/SeedVaultExportComponent.js [hunk 1]
```diff
@@ -298,7 +298,7 @@ class SeedVaultExportComponent extends Component {
         const textColor = { color: theme.body.color };
 
         return (
-            <Animated.View style={[styles.container, { transform: [{ translateX: this.animatedValue }] }]}>
+            <Animated.View style={[styles.container, { transform: [{ withNamespacesX: this.animatedValue }] }]}>
```
Comment (rajivshah3) [inline src/mobile/src/ui/components/SeedVaultExportComponent.js]: `translateX` is a prop from https://facebook.github.io/react-native/docs/transforms#transform so it shouldn't be changed

**Review by rajivshah3 — DISMISSED** [review]

File: src/desktop/package.json [hunk 2]
```diff
@@ -133,27 +132,27 @@
     "bugsnag-js": "^4.7.3",
     "bugsnag-react": "^1.1.1",
     "classnames": "^2.2.6",
-    "curl.lib.js": "^1.0.22",
     "electron-settings": "^3.2.0",
     "electron-updater": "^3.1.2",
     "entangled-node": "rihardsgravis/entangled-node",
+    "fsevents": "1.2.4",
```
Comment (rajivshah3) [inline src/desktop/package.json]: What is `fsevents` for? It seems to be causing the CI tests to fail

File: src/desktop/package.json [hunk 3]
```diff
@@ -133,27 +132,27 @@
...
     "perfect-scrollbar": "^1.4.0",
     "qr.js": "^0.0.0",
     "react": "^16.5.2",
     "react-dom": "^16.5.2",
-    "react-i18next": "^7.12.0",
+    "react-i18next": "^8.0.7",
     "react-lottie": "^1.2.3",
-    "react-markdown": "^3.6.0",
-    "react-qr-reader": "^2.1.0",
+    "react-markdown": "^4.0.2",
```
Comment (rajivshah3) [inline src/desktop/package.json]: `react-markdown` shouldn't be updated past 3.6.0 unless absolutely necessary. Later versions introduce a vulnerable dependency (https://app.snyk.io/test/npm/react-markdown/4.0.2)

File: src/desktop/package.json (hunk 2 above)
Comment (rihardsgravis) [reply src/desktop/package.json]: fsevents is used at kdbxweb, but an older version that fails to build on Node 10. Removed this dependency and added a correct shrinkwrap file.

**Review by rajivshah3 — APPROVED** [review]

**Comment by rajivshah3** [comment]
~Blocked by https://github.com/iotaledger/ci-pipelines/pull/8~

**Comment by rajivshah3** [comment]
@rihardsgravis I'm seeing this in the [Buildkite logs](https://buildkite.com/iota-foundation/trinity-desktop-prs/builds/729#2643bfae-bddf-4aeb-bd44-d858831f2efa/21-92):

```
  | npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
  | npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.4 (node_modules/kdbxweb/node_modules/fsevents):
  | npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
```

Is this normal?

**Pull request merged**